framework/src/modules/interoperability/base_interoperability_store.ts

```diff
@@ -241,65 +244,72 @@ export abstract class BaseInteroperabilityStore {
 		return terminatedOutboxSubstore.get(this.context, chainID);
 	}
 
-	public async createTerminatedStateAccount(chainID: Buffer, stateRoot?: Buffer): Promise<boolean> {
-		const chainSubstore = this.stores.get(ChainAccountStore);
-		const isExist = await this.chainAccountExist(chainID);
+	public async createTerminatedStateAccount(
+		context: CreateTerminatedStateAccountContext,
+		chainID: Buffer,
+		stateRoot?: Buffer,
+	): Promise<void> {
 		let terminatedState: TerminatedStateAccount;
 
-		if (stateRoot) {
-			if (isExist) {
-				const chainAccount = await chainSubstore.get(this.context, chainID);
-				chainAccount.status = CHAIN_TERMINATED;
-				await chainSubstore.set(this.context as StoreGetter, chainID, chainAccount);
-				const outboxRootSubstore = this.stores.get(OutboxRootStore);
-				await outboxRootSubstore.del(this.context as StoreGetter, chainID);
-			}
-			terminatedState = {
-				stateRoot,
-				mainchainStateRoot: EMPTY_BYTES,
-				initialized: true,
-			};
-		} else if (isExist) {
+		const chainSubstore = this.stores.get(ChainAccountStore);
+		const chainAccountExists = await this.chainAccountExist(chainID);
```

Comment:
you can use `chainSubstore.has` instead of using store's chainAccountExist function as we want to remove it in the coming PR